packages/rest/src/lib/REST.ts

```diff
@@ -434,16 +434,20 @@ export class REST extends AsyncEventEmitter<RestEventsMap> {
 			};
 		}
 
-		const majorIdMatch = /^\/(?:channels|guilds|webhooks)\/(\d{17,19})/.exec(endpoint);
+		const majorIdMatch = /(?:^\/webhooks\/(\d{17,19}\/[^/]+))|(?:^\/(?:channels|guilds|webhooks)\/(\d{17,19}))/.exec(
```

Comment:
```suggestion
		const majorIdMatch = /(?:^\/webhooks\/(\d{17,19}\/[^/?]+))|(?:^\/(?:channels|guilds|webhooks)\/(\d{17,19}))/.exec(
```

Shouldn't match query params either